src/pages/task/taskTabs/FileTableTab/GroupedFilesTable/index.tsx

```diff
@@ -0,0 +1,96 @@
+import { useMemo, useRef } from "react";
+import {
+  Table,
+  TableHead,
+  HeaderRow,
+  HeaderCell,
+  TableBody,
+  Row,
+  Cell,
+  LGColumnDef,
+  useLeafyGreenTable,
+  HeaderGroup,
+  LeafyGreenTableRow,
+  flexRender,
+} from "@leafygreen-ui/table/new";
+import { Subtitle } from "@leafygreen-ui/typography";
+import { StyledLink } from "components/styles";
+import { Unpacked } from "types/utils";
+import { GroupedFiles } from "../types";
+
+type GroupedFilesFile = Unpacked<GroupedFiles["files"]>;
+interface GroupedFilesTableProps {
+  files: GroupedFilesFile[];
+  taskName?: string;
+}
+const GroupedFilesTable: React.FC<GroupedFilesTableProps> = ({
+  files,
+  taskName,
+}) => {
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+
+  const columns = useMemo<LGColumnDef<GroupedFilesFile>[]>(
+    () => [
+      {
+        accessorKey: "name",
+        header: "Name",
+        size: 60,
+        enableSorting: true,
+        cell: (value) => (
+          <StyledLink
+            href={value.row.original.link}
+            data-cy="file-link"
+            target="_blank"
+          >
+            {value.getValue()}
+          </StyledLink>
+        ),
+      },
+    ],
+    []
+  );
+  const table = useLeafyGreenTable<GroupedFilesFile>({
+    containerRef: tableContainerRef,
+    data: files,
+    columns,
+  });
+
+  const { rows } = table.getRowModel();
+
+  return (
+    <>
+      {taskName && <Subtitle>{taskName}</Subtitle>}
+      <Table ref={tableContainerRef}>
```

Comment:
Could you possibly use the [`BaseTable`](https://github.com/evergreen-ci/spruce/blob/main/src/components/Table/BaseTable.tsx) component here?

Reply:
Great call!